crates/cext/src/transpiler/target.rs

```diff
@@ -869,6 +872,474 @@ pub unsafe extern "C" fn qk_target_num_instructions(target: *const Target) -> us
+/// Behavior is undefined if ``name`` is not a pointer to a valid null-terminated string.
+#[unsafe(no_mangle)]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_target_op_get_index(
+    target: *const Target,
+    name: *const c_char,
+) -> usize {
+    // SAFETY: Per documentation, the pointer is non-null and aligned.
+    let target_borrowed = unsafe { const_ptr_as_ref(target) };
+    // SAFETY: Per documentation, this should point to a valid, null-terminated
+    // string.
+    let name = unsafe {
+        CStr::from_ptr(name)
+            .to_str()
+            .expect("Error extracting gate name.")
+    };
+
+    if let Some(index) = target_borrowed.get_gate_index(name) {
+        index
+    } else {
+        usize::MAX
+    }
+}
+
+/// @ingroup QkTarget
+/// Return the name of the operation stored at that index in the ``QkTarget`` instance's
+/// gate map.
+///
+/// @param target A pointer to the ``QkTarget``.
+/// @param index The index in which the gate is stored.
+///
+/// # Example
+/// ```c
+///     QkTarget *target = qk_target_new(5);
+///     QkTargetEntry *target_entry = qk_target_entry_new(QkGate_H);
+///     qk_target_add_instruction(target, target_entry);
+///
+///     char *op_name = qk_target_op_name_from_index(target, 0);
+///     // Free after use
+///     qk_str_free(op_name);
+/// ```
+///
+/// # Safety
+///
+/// Behavior is undefined if ``QkTarget`` is not a valid, non-null pointer to a ``QkTarget``.
+#[unsafe(no_mangle)]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_target_op_name_from_index(
+    target: *const Target,
+    index: usize,
+) -> *mut c_char {
+    // SAFETY: Per documentation, the pointer is non-null and aligned.
+    let target_borrowed = unsafe { const_ptr_as_ref(target) };
+    if let Some((retrieved, _)) = target_borrowed.get_from_index(index) {
+        CString::new(retrieved.clone())
+            .expect("Error allocating space for string")
+            .into_raw()
+    } else {
+        panic!("Index does not map to any operation in the Target.")
+    }
+}
+
+/// @ingroup QkTarget
+/// Checks whether an operation is present in the ``QkTarget`` instance by index.
+///
+/// @param target A pointer to the ``QkTarget``.
+/// @param name A pointer to the name string which we will look up.
+///
+/// # Example
+/// ```c
+///     QkTarget *target = qk_target_new(5);
+///     QkTargetEntry *target_entry = qk_target_entry_new(QkGate_H);
+///     qk_target_add_instruction(target, target_entry);
+///
+///     if (qk_target_contains_op(target, "h")) {
+///         printf("'h' has been added to the Target.");
+///     };
+/// ```
+///
+/// # Safety
+///
+/// Behavior is undefined if ``QkTarget`` is not a valid, non-null pointer to a ``QkTarget``.
+/// Behavior is undefined if ``name`` is not a pointer to a valid null-terminated string.
+#[unsafe(no_mangle)]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_target_contains_op(target: *const Target, name: *const c_char) -> bool {
```

Comment:
Just a suggestion, but perhaps we should call this `qk_target_contains` instead to reserve the word "op" for functions that use an index?